mappings/net/minecraft/entity/passive/VillagerEntity.mapping

```diff
@@ -1,39 +1,59 @@
 CLASS aqr net/minecraft/entity/passive/VillagerEntity
 	FIELD bD VILLAGER_TYPE Lpr;
 	FIELD bH properties Lalq;
-	FIELD bJ recipeList Lbav;
+	FIELD bI tradingPlayer Larb;
+	FIELD bJ offers Lbav;
+	FIELD bM willing Z
+	FIELD bN lastTradedWith Ljava/lang/String;
 	FIELD bQ inventory Lafx;
 	METHOD <init> (Lbbp;Laqw;)V
 		ARG 1 world
 	METHOD D getAmbientSound ()Lxm;
 	METHOD I mobTick ()V
 	METHOD P_ getCurrentCustomer ()Larb;
 	METHOD Q getDisplayName ()Ljd;
+	METHOD Q_ getOngoingRaid ()Lasf;
 	METHOD a onDeath (Lage;)V
+		ARG 1 damageSource
+		ARG 2 attacker
+		ARG 3 nearbyPlayer
 	METHOD a createChild (Lags;)Lags;
 	METHOD a setAttacker (Lahe;)V
+		ARG 1 attacker
 	METHOD a onStruckByLightning (Laov;)V
+		ARG 1 lightning
+		ARG 2 witch
 	METHOD a setVillagerType (Laqs;)V
 	METHOD a canBeLeashedBy (Larb;)Z
+		ARG 1 player
 	METHOD a interactMob (Larb;Lafo;)Z
 		ARG 1 player
 	METHOD a onSellingItem (Lawo;)V
-	METHOD a useRecipe (Lbau;)V
-	METHOD a setRecipeList (Lbav;)V
+	METHOD a useOffer (Lbau;)V
+		ARG 1 offer
+		ARG 2 exp
+	METHOD a setOffers (Lbav;)V
+		ARG 1 offers
 	METHOD a (Lbbq;Lafn;Lahh;Lahp;Lhs;)Lahp;
 		ARG 3 difficulty
+	METHOD a spawnParticle (Lft;)V
+		ARG 1 particle
 	METHOD a readCustomDataFromTag (Lhs;)V
 	METHOD a_ setCurrentCustomer (Larb;)V
 	METHOD b createChild (Lags;)Laqr;
 	METHOD b writeCustomDataToTag (Lhs;)V
 	METHOD bG getEyeHeight ()F
-	METHOD c getRecipes ()Lbav;
+	METHOD c getOffers ()Lbav;
 	METHOD ca initAttributes ()V
 	METHOD cv getDeathSound ()Lxm;
 	METHOD d getVillagerWorld ()Lbbp;
+	METHOD dB isTrading ()Z
 	METHOD dD getInventory ()Lafx;
+	METHOD dJ ()V
 	METHOD dM ()Laqs;
 	METHOD e getVillagerPos ()Let;
 	METHOD e getHurtSound (Lage;)Lxm;
 	METHOD n ()V
+	METHOD v setWilling (Z)V
```

Comment:
Could this be a tad more descriptive?

Reply:
I mean the field and nbt tag are called willing